content/docs/manual/access-control.md

```diff
@@ -0,0 +1,201 @@
+        ],
+        "usernames": [
+          "zenoh-example1",
+          "zenoh-example2",
+        ],
+      }
+    ],
+  
+    "policies": [
+      {
+        "rules": [
+          "deny pub/sub"
+        ],
+        "subjects": [
+          "example subject"
+        ],
+      },
+    ]
+  }
+}
+```
+
+### rules
+
+Each rule within the `rules` list is identified by a unique `id` string. Rules apply on matched messages based on their individual characteristics:
+
+- `messages`: types of messages to apply the rule on. Supported message types are the following:
+  - Declare Subscriber (`declare_subscriber`)
+  - Publication (`put`)
+  - Delete (`delete`)
+  - Declare Queryable (`declare_queryable`)
+  - Query (`query`)
+  - Reply (`reply`)
+- `flows`: applies rule on incoming messages (`ingress`), outgoing messages (`egress`), or both directions. If this field is not provided in the config, the rule will apply to both directions by default.
+- `key_exprs`: the rule applies on messages for which the key matches one of the given key expressions. For more details on key expression matching, see [Key Expressions](http://localhost:1313/docs/manual/abstractions/#key-expression).
+
+Matched messages are filtered based on the rule's `permission`: `allow` or `deny`.
+
+For instance, the following rule denies all incoming and outgoing subscriptions, publications and deletions on key expressions matching `demo/example/**`:
+
+```json
+{
+  "id": "deny pub/sub",
+  "permission": "deny",
+  "flows": ["ingress", "egress"],
+  "messages": [
+    "declare_subscriber",
+    "put",
+    "delete",
+  ],
+  "key_exprs": [
+    "demo/example/**",
+  ],
+}
+```
+
+### subjects
+
+Each subject configuration is identified by a unique `id` string. Subject configuration combines `interfaces`, `cert_common_names` and `usernames` to match with configurations of connecting Zenoh instances.
+
+- `interfaces`: list of local network interfaces through which the configured instance communicates with the remote instance to be matched.
+- `cert_common_names`: list of certificate common names which are matched with the respective certificate content of the remote instance using TLS or QUIC transport. For details on certificate configuration refer to [TLS authentication](../tls) and [QUIC transport](../quic).
+- `usernames`: list of usernames to be matched with the authentication config of remote instances. Refer to [User-Password authentication](../user-password) for how to setup this authentication mechanism.
+
+To produce all possible combinations that characterize a subject configuration, the Cartesian product of the `interfaces`, `cert_common_names` and `usernames` lists is calculated. This allows for items within the same list to be considered a logical `OR`, and items across different lists to be considered a logical `AND`.
+To demonstrate these logical combinations, below is an example of a subject configurations and its internal representation.
```

Comment:
"below is an example of a subject configuration" 
either plural subject configurations or singular configuration.

Reply:
The s is a typo, it's singular (one subject configuration).